test/Interop/Cxx/objc-correctness/Inputs/NSOptionsMangling.h

```diff
@@ -0,0 +1,50 @@
+#define __CF_OPTIONS_ATTRIBUTES __attribute__((flag_enum,enum_extensibility(open)))
+#if (__cplusplus)
+#define CF_OPTIONS(_type, _name) __attribute__((availability(swift,unavailable))) _type _name; enum __CF_OPTIONS_ATTRIBUTES : _name
+#else
+#define CF_OPTIONS(_type, _name) enum __CF_OPTIONS_ATTRIBUTES _name : _type _name; enum _name : _type
+#endif
+
+
+#define NS_OPTIONS(_type, _name) CF_OPTIONS(_type, _name)
+typedef unsigned long NSUInteger;
+typedef NS_OPTIONS(NSUInteger, UIControlState) {
+    UIControlStateNormal       = 0,
+    UIControlStateHighlighted  = 1 << 0,                  // used when UIControl isHighlighted is set
+};
+
+#define NS_SWIFT_UI_ACTOR __attribute__((swift_attr("@UIActor")))
+#define __API_AVAILABLE_PLATFORM_macos(x) macos,introduced=x
+#define __API_AVAILABLE_PLATFORM_macosx(x) macosx,introduced=x
+#define __API_AVAILABLE_PLATFORM_ios(x) ios,introduced=x
+#define __API_AVAILABLE_PLATFORM_watchos(x) watchos,introduced=x
+#define __API_AVAILABLE_PLATFORM_tvos(x) tvos,introduced=x
+#define __API_AVAILABLE_PLATFORM_macCatalyst(x) macCatalyst,introduced=x
+#define __API_AVAILABLE_PLATFORM_macCatalyst(x) macCatalyst,introduced=x
+#define __API_AVAILABLE_PLATFORM_uikitformac(x) uikitformac,introduced=x
+#define __API_AVAILABLE_PLATFORM_driverkit(x) driverkit,introduced=x
+#define __API_A(x) __attribute__((availability(__API_AVAILABLE_PLATFORM_##x)))
+#define __API_AVAILABLE1(x) __API_A(x)
+#define __API_AVAILABLE2(x,y) __API_A(x) __API_A(y)
+#define __API_AVAILABLE3(x,y,z)  __API_A(x) __API_A(y) __API_A(z)
+#define __API_AVAILABLE4(x,y,z,t) __API_A(x) __API_A(y) __API_A(z) __API_A(t)
+#define __API_AVAILABLE5(x,y,z,t,b) __API_A(x) __API_A(y) __API_A(z) __API_A(t) __API_A(b)
+#define __API_AVAILABLE6(x,y,z,t,b,m) __API_A(x) __API_A(y) __API_A(z) __API_A(t) __API_A(b) __API_A(m)
+#define __API_AVAILABLE7(x,y,z,t,b,m,d) __API_A(x) __API_A(y) __API_A(z) __API_A(t) __API_A(b) __API_A(m) __API_A(d)
+#define __API_AVAILABLE8(x,y,z,t,b,m,d,l) __API_A(x) __API_A(y) __API_A(z) __API_A(t) __API_A(b) __API_A(m) __API_A(d) __API_A(l)
+#define __API_AVAILABLE_GET_MACRO(_1,_2,_3,_4,_5,_6,_7,_8,NAME,...) NAME
+#define API_AVAILABLE(...) __API_AVAILABLE_GET_MACRO(__VA_ARGS__,__API_AVAILABLE8, __API_AVAILABLE7, __API_AVAILABLE6, __API_AVAILABLE5, __API_AVAILABLE4, __API_AVAILABLE3, __API_AVAILABLE2, __API_AVAILABLE1, 0)(__VA_ARGS__)
```

Comment:
Is this whole thing necessary for the test?

Reply:
You are right, it is not. First draft I just wanted to put up the most complete set of macros used so that it reproduces. I will update the PR to clean things up.